lib/tests/run_tests.js

```diff
@@ -123,6 +124,23 @@ module.exports = {
             eachCb();
           });
         }, next);
+      },
+      function runCoverage(next) {
+        if (!options.coverage) {
+          return next();
+        }
+
+        runCmd(`${fs.embarkPath('node_modules/.bin/istanbul')} report --root .embark --format html`,
+          {silent: false, exitOnError: false}, (err) => {
+            if (err) {
+              return next(err);
+            }
+            console.log(`Coverage report created. You can find it here: ${fs.dappPath('coverage/index.html')}\n`);
+            const opn = require('opn');
+            opn(fs.dappPath('coverage/index.html'));
+
+            setTimeout(next, 1000); // Slight delay to enable the browser to open. opn's promise calls too fast
```

Comment:
`opn()` returns a promise.

See: https://github.com/embark-framework/embark/blob/833fcef22ae53dd55a63d78a4bf4d7f356896fc3/lib/modules/webserver/index.js#L134L137

It's important to use the `{wait: false}` option and to call `next` even if opn fails (it will fail in a docker container, for example).

Reply:
Like I said in the comment, the promise returned immediately, without waiting for the open.
I think it's a Windows issue, but still

Reply:
Did you try it with `{wait: false}`? The default for `wait:` is `true`, and that is known to cause a problem on Windows if an app isn't explicitly specified.

Reply:
I did

Reply:
Okay, fair enough, but maybe it's still important to use `.then(() => { next(); }, () => { next(); })`?  Won't we get an unhandled promise rejection error if `opn()` fails.

In that case, it would be important to specify `{wait: false}` for the `opn()` call because we don't actually want to (try to) wait for the browser to exit before calling `next`.

The delay you've found we need could be intro'd with an async function:

```
opn(...)
  .then(async () => await (new Promise(resolve => setTimeout(resolve, 1000))))
  .then(() => { next(); }, () => { next(); });
```

Reply:
With the PR as is, when running `embark test --coverage` for a fresh `embark_demo` inside the docker container:

```
embark test --coverage
Compiling contracts


  SimpleStorage
    ✓ should set constructor value (92ms) - [21666 gas]
    ✓ set storage value (119ms) - [48409 gas]
    ✓ should have account with balance (2ms) - [0 gas]


  3 passing (5s) - [Total: 220332 gas]

Done

Coverage report created. You can find it here: /home/embark/dapps/embark_demo/coverage/index.html

(node:2660) UnhandledPromiseRejectionWarning: Error: Exited with code 3
    at ChildProcess.cp.once.code (/home/embark/repos/embark/node_modules/opn/index.js:84:13)
    at Object.onceWrapper (events.js:317:30)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at maybeClose (internal/child_process.js:915:16)
    at Socket.stream.socket.on (internal/child_process.js:336:11)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at Pipe._handle.close [as _onclose] (net.js:561:12)
(node:2660) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:2660) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
 > All tests passed
```

When I revised `lib/tests/run_tests.js ` with the code block at the bottom of my previous comment, `UnhandledPromiseRejectionWarning` goes away.

Reply:
Oh ok then, my bad

Reply:
I'm testing my suggested revisions w/ Windows, will let you know if I find any problems in a few minutes.

Reply:
It took forever to get my new Win VM setup w/ `windows-build-tools`, apologies for the slow reply. One thing I've found is that on a slow Windows computer (like my VM) 1 second isn't enough of a delay. However, I don't think we should we increase the delay, since I think the need for it is essentially a bug in the `opn` package with respect to Windows. On macOS and desktop Linux, as far as I can tell, the `opn` call always works even if there's no delay.